src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/modifier/interfacememberimpliedmodifier/Example1.java

```diff
@@ -0,0 +1,42 @@
+/*xml
+<module name="Checker">
+  <module name="TreeWalker">
+    <module name="InterfaceMemberImpliedModifier"/>
+  </module>
+</module>
+*/
+package com.puppycrawl.tools.checkstyle.checks.modifier.interfacememberimpliedmodifier;
+
+import java.util.List;
+
+// xdoc section -- start
+class Example1 {
+  public interface AddressFactory {
```

Comment:
Can we avoid wrapper class?
And make 
`public interface Example1 {`

Reply:
What is a violation?

Reply:
` Example test input must declare exactly one class matching 'Example\d+' [exampleClassName]`

Reply:
Please update https://github.com/checkstyle/checkstyle/blob/31dbc43ba0e8baf15872c0c62fc379197fb1d419/config/checkstyle-examples-checks.xml#L123

To allow `interface` and `record` and `enum`

Reply:
Done